deployment: point out intended workflow

yogo1212 opened the pull request:
### What does it do?

Admins are confused, contractors create content on their dev machines and are unhappy that they have to redo their work on prod... Just let people know.

### Why is it needed?

Devops will try ludicrous things in order to get the content from developers into production.
Tell them that's not something they should be doing :-)

Boegie19 commented:
@yogo1212 Thanks for the PR but I am not sure if deployment is the right location for this warning 

I think it would make more sense to split the warning into 2. add content on the production instance for the user guide

And the development part under FAQ and and under quick start guide some text to explain the same thing.

lets see what @pwizla says since he is the maintainer here.

yogo1212 commented:
being such an admin who received a repo and the instructions to bring it online, i'm looking at the deployment page but i totally agree
i've had two devs who said they were experienced with strapi give me their content exports while there was a production instance of strapi up.
so i would be happy to have some clear, dev-facing documentation to point them to as well.
judging by the amount of forum posts, questions on SO and discord, i'm not the only one admin with the problem. - the more places the better.

pwizla commented:
Thanks, @yogo1212. I think adding this piece of information here makes sense as some best practice recommendations. However, this is just one of the possible workflows, and one can also use the [data management system](https://docs.strapi.io/dev-docs/data-management) to export data, import data, or even transfer data from a Strapi instance to another one. 
As it's easier/faster for me, I will close your PR and directly push an update to the `main` branch.
Thank you!

yogo1212 commented:
@pwizla thanks for your work and for pointing out those additional methods!

we were hoping to use the import on the second attempt (the first dev just gave us a sqlite file and a tar archive..) but it overrides all existing data or configuration.
that means it's really only suited for deploying staging to prod.
any workflow involving multiple content creators or existing data excludes the use of import/export or transfer.

imho, the documentation you have linked could also benefit from a note pointing that out.
would a PR adding such a note to the [data management page](https://docs.strapi.io/dev-docs/data-management) be welcome?
(maybe: `All of these methods delete the data on the target. Moving data between instances that are actively used for adding content is not intended.`)

pwizla commented:
Thanks for your feedback and additional questions, @yogo1212. I'm sorry you've had issues with the data management system.
Could you please explain what is your use case "involving multiple content creators"?

Also, by "all of these methods delete the data on the target", this is mentioned when you run the CLI command and you can still refuse to launch it. I'm trying to understand your need: would you like existing data to be merged with the imported content? 🤔 

The data import, export, and transfer features allow including only some types of elements (files, content, or configuration) with the `--only` flag or excluding them with the `--exclude` flag.

Have you considered trying this? If it still does not solve your issue, I would encourage you to [create an issue on the strapi/strapi repository](https://github.com/strapi/strapi/issues), and we'll also what we can do on our side.

yogo1212 commented:
@pwizla Thank you for your response :-) No reason to be sorry - let's just check if there's a way for improvement.

> Could you please explain what is your use case "involving multiple content creators"?

Short version: Multiple blog post contributors on prod, one or more developers sending their exports.

At first, I had just configured the 'bare' Strapi instance, created users, configured nodemailer, and such. All of that worked just as documented.
A contractor was supposed to create the website from Figma templates and we (including the Strapi expert) assumed that import/export/transfer were reasonable ways to import his work into the production instance.
After types were created and deployed, content was added on prod by two other people.
The contractor continued working on his dev instance, changed models, and added content.
Only then did we realize that it wouldn't be possible to apply his content on prod without him having to re-do all the work or us having to re-do all of our work (he still doesn't understand that he shouldn't be adding content on his devel clone which will likely be the reason for a cancellation of his contract).

Imo, it would have been useful to know this in advance.

Boegie19 commented:
Only way how want you want is currently possible is https://github.com/Baboo7/strapi-plugin-import-export-entries . But this plugin can be janky and crash your admin with updates

pwizla commented:
Thanks for the details, @yogo1212! 
As this could be a documentation improvement as much as a product-related question, have you or your contractors already considered contacting the [support](https://docs.strapi.io/dev-docs/intro#support) or creating an [issue on strapi/strapi](https://github.com/strapi/strapi/issues) (not sure if you're on a free or Enterprise plan)? 
I think that [Strapi Cloud](https://strapi.io/cloud) might be the answer. For now, if several devs are working on the repo deployed to Strapi Cloud, they would have to use a unique, shared GitHub account, but I'm happy to announce that shared project access on Strapi Cloud is in the works (it was demo-ed internally yesterday) :) The multiple blog post contributors on prod shouldn't be an issue with [Strapi Cloud](https://strapi.io/pricing-cloud), and the [Review Workflows](https://docs.strapi.io/user-docs/settings/review-workflows) features could also help the various contributors.

If you create a support issue on strapi/strapi, please feel free to ping me so I can follow along :)
